assembl/fabfile.py

```diff
@@ -794,6 +801,155 @@ def update_pip_requirements(force_reinstall=False):
         run("yes w | %s" % cmd)
 
 
+def is_db_updated():
+    """
+    Return if the database is update or not
+    """
+    history = venvcmd('alembic -c %s history' % (env.ini_file))
+    current = venvcmd('alembic -c %s heads' % (env.ini_file))
+    return current in history
+
+
+@task
+def reset_db():
+    """
+    Restore and update the latest database
+    """
+    # Only for the staging server (for tests)
+    if env.wsginame == 'staging.wsgi':
+        exists = False
+        # Test if the dump exists on the amazon s3 object storage
+        with shell_env(
+            AWS_ACCESS_KEY_ID=env.aws_access_key_id,
+            AWS_SECRET_ACCESS_KEY=env.aws_secret_access_key
+        ):
+            exists = venvcmd('aws s3 ls s3://%s/%s | wc -l' % (env.aws_bucket_name, get_db_dump_name()))
+            exists = False if exists == '0' else True
+
+        print(green('Restore and update the latest database'))
+        if not exists:
+            # If the dump don't exist, we create it
+            print(cyan('Create a new dump'))
+            execute(database_dump_aws)
+        else:
+            # Otherwise, we restore the last dump
+            print(cyan('Restore the last dump'))
+            execute(database_restore_aws)
+
+        # Update the dump only when the schema of the database changes
+        if not is_db_updated():
+            print(cyan('Update the restored db'))
+            execute(app_db_update)
+            # Create the updated dump for future tests
+            print(cyan('Create the updated dump for future tests'))
+            execute(database_dump_aws)
+
+
+@task
+def database_dump_aws():
+    """
+    Dumps the database on an amazon s3 object storage
+    """
+    with prefix(venv_prefix()), \
+        cd(env.projectpath), \
+        shell_env(
+        AWS_ACCESS_KEY_ID=env.aws_access_key_id,
+        AWS_SECRET_ACCESS_KEY=env.aws_secret_access_key
+    ):
+        dump_name = get_versioned_db_dump_name()
+        dump_path = os.path.join(env.dbdumps_dir, dump_name)
+        # Create the db dump
+        run('PGPASSWORD=%s pg_dump --host=%s -U%s --format=custom -b %s > %s' % (
+            env.db_password,
+            env.db_host,
+            env.db_user,
+            env.db_database,
+            dump_path))
+        # Copy the created dump in the aws bucket
+        run('aws s3 cp %s s3://%s/%s ' % (
+            dump_path,
+            env.aws_bucket_name,
+            dump_name))
+        # Add a copy as a symbolic link
+        run('aws s3 cp %s s3://%s/%s ' % (
+            dump_path,
+            env.aws_bucket_name,
+            get_db_dump_name()))
+        # Remove the created dump from the locale host
+        run('rm -f %s' % dump_path)
+
+
+@task
+def database_restore_aws(backup=False):
+    """
+    Restores the database backed up on the amazon s3 object storage
+    """
+    if not backup:
+        assert(env.wsginame in ('staging.wsgi', 'dev.wsgi'))
```

Comment:
what's the difference between staging and dev I thought there is only one host?

Reply:
it's for the local dev